src/libraries/System.IO.FileSystem.AccessControl/src/System/IO/FileSystemAclExtensions.cs

```diff
@@ -200,18 +195,33 @@ public static FileStream Create(this FileInfo fileInfo, FileMode mode, FileSyste
                 throw new ArgumentOutOfRangeException(nameof(bufferSize), SR.ArgumentOutOfRange_NeedPosNum);
             }
 
-            // Do not allow using combinations of non-writing file system rights with writing file modes
+            // Do not combine writing modes with exclusively read rights
+            // Write contains AppendData, WriteAttributes, WriteData and WriteExtendedAttributes
             if ((rights & FileSystemRights.Write) == 0 &&
                 (mode == FileMode.Truncate || mode == FileMode.CreateNew || mode == FileMode.Create || mode == FileMode.Append))
             {
                 throw new ArgumentException(SR.Format(SR.Argument_InvalidFileModeAndFileSystemRightsCombo, mode, rights));
             }
 
+            // Additionally, append is disallowed if any read rights are provided
+            // ReadAndExecute contains ExecuteFile, ReadAttributes, ReadData, ReadExtendedAttributes and ReadPermissions
+            if ((rights & FileSystemRights.ReadAndExecute) != 0 && mode == FileMode.Append)
+            {
+                throw new ArgumentException(SR.Argument_InvalidAppendMode);
+            }
+
+            // Cannot truncate unless all of the write rights are provided
+            // Write contains AppendData, WriteAttributes, WriteData and WriteExtendedAttributes
+            if (mode == FileMode.Truncate && (rights & FileSystemRights.Write) != FileSystemRights.Write)
+            {
+                throw new ArgumentException(SR.Format(SR.Argument_InvalidFileModeAndFileSystemRightsCombo, mode, rights));
+            }
```

Comment:
Here's the code that verifies this in .NET Framework:
https://referencesource.microsoft.com/#mscorlib/system/io/filestream.cs,688

Reply:
By the way, the successful case for this check is the one with the `ActiveIssue` attribute in the tests. The `CreateFile` P/Invoke is who's throwing the exception. I'm investigating.